build/parsePreamble.c

```diff
@@ -1248,7 +1257,29 @@ int parsePreamble(rpmSpec spec, int initialPackage)
 				spec->lineNum, spec->line);
 		    goto exit;
 		}
-		if (handlePreambleTag(spec, pkg, tag, macro, lang)) {
+		if (stage == GENERATED) {
+		    switch (tag) {
+		    case RPMTAG_SOURCE:
+		    case RPMTAG_PATCH:
+		    case RPMTAG_NOSOURCE:
+		    case RPMTAG_NOPATCH:
+		    case RPMTAG_EXCLUDEARCH:
+		    case RPMTAG_EXCLUSIVEARCH:
+		    case RPMTAG_EXCLUDEOS:
+		    case RPMTAG_EXCLUSIVEOS:
+		    case RPMTAG_BUILDROOT:
+		    case RPMTAG_BUILDCONFLICTS:
+		    case RPMTAG_BUILDOPTION:
+		    case RPMTAG_BUILDPREREQ:
+		    case RPMTAG_BUILDREQUIRES:
+		    case RPMTAG_BUILDSYSTEM:
```

Comment:
Add this info to the struct in preambleList instead, that forces one to consider this issue whenever new tags are added.
You can check for this from findPreambleTag() just as well.

Reply:
Done.